### PR TITLE
Add support for command line args in test schemes

### LINKED ISSUE #1514: Feature Request: Support `Arguments Passed On Launch` in `ios_unit_test`
Currently in `bazel test` it is possible to set something like `args = ["--command_line_args=-AppleLanguages,(en)"],` in order to pass `Arguments Passed On Launch` for test scheme. 
Expected Output would be: `-AppleLanguages (en)` (space separated)

It would be very useful to have same feature in `rules_xcodeproj` in order to achieve something like:

image

Regarding `--command_line_args`, it is [from here](https://github.com/bazelbuild/rules_apple/pull/1161).

### PR DESCRIPTION
**Pull request by maxwellE**
Support for passing command line args to
test schemes

Fixes #1514

### CODE REVIEW

File: tools/generator/src/DTO/Project.swift [hunk 1]
```diff
@@ -11,6 +11,7 @@ struct Project: Equatable {
     let replacementLabels: [TargetID: BazelLabel]
     let targetHosts: [TargetID: [TargetID]]
     let envs: [TargetID: [String: String]]
+    let args: [TargetID: [String]]
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Project.swift:14]: nit: can you have `args` above `env` everywhere they both exist?

Comment (maxwellE) [reply tools/generator/src/DTO/Project.swift:14]: Will do

File: tools/generator/src/Extensions/Array+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
+extension Array where Element == String {
+    func extractCommandLineArguments() -> [String] {
+        let validCommandLineArgSuffix: String = "--command_line_args="
```
Comment (brentleyjones) [inline tools/generator/src/Extensions/Array+Extensions.swift:3]: Can you pull this into a `static let`?

File: examples/integration/iOSApp/Test/SwiftUnitTests/BUILD [hunk 3]
```diff
@@ -3,6 +3,7 @@ load("@build_bazel_rules_swift//swift:swift.bzl", "swift_library")
 
 ios_unit_test(
     name = "iOSAppSwiftUnitTests",
+    args = ["--command_line_args=-AppleLanguages,(en)"],
```
Comment (brentleyjones) [inline examples/integration/iOSApp/Test/SwiftUnitTests/BUILD:6]: How does this interact with custom scheme args? I would assume the custom scheme ones override these?

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
@rasberik can you confirm this PR addressed your issue? Thanks!

File: examples/integration/iOSApp/Test/SwiftUnitTests/BUILD (hunk 3 above)
Comment (maxwellE) [reply examples/integration/iOSApp/Test/SwiftUnitTests/BUILD:6]: I will add a test case for this scenario

**Review by brentleyjones — APPROVED** [review]
Thanks @maxwellE!

I'm going to wait to merge until I get confirmation from @rasberik.

**Comment by rasberik** [comment]
```args = ["--command_line_args=-AppleLanguages,(en)"]``` (comma separated)
Expected Output would be: `-AppleLanguages (en)` (space separated)

Also, I noticed from [original source](https://github.com/bazelbuild/rules_apple/pull/1161/files) that there can be several `--command_line_args`, so each new argument is defined separately, and comma representing space?

@ra1028 Can you please have a look if you have time?

**Comment by ra1028** [comment]
In my understanding, this PR already supports extracting multiple --command_line_args in `Array.extractCommandLineArguments()`. However, rules_apple also supports a comma-separated list of arguments in a single flag, so I think rules_xcodeproj should follow that too.

e.g.
`--command_line_args=-UIPreferredContentSizeCategoryName,UICTContentSizeCategoryXXXL`
```
- -UIPreferredContentSizeCategoryName
- UICTContentSizeCategoryXXXL
```

If you pass space-separated arguments in Xcode and dump `ProcessInfo.processInfo.arguments`, you'll see that they are propagated as separate elements like `["-AppleLanguages", "(ja)"]`.
It means that passing the following two results in the same.

```
- -AppleLanguages (ja)
```
```
- -AppleLanguages
- (ja)
```

Therefore, I think @rasberik's concerns would solve if comma-separated arguments will be supported.

File: tools/generator/src/Extensions/Array+Extensions.swift [hunk 4]
```diff
@@ -3,12 +3,12 @@ extension Array where Element == String {
...
     func extractCommandLineArguments() -> [String] {
-        return compactMap({ arg -> String? in
+        return flatMap({ arg -> [String] in
             guard arg.hasPrefix(Array.validCommandLineArgSuffix)
             else {
-                return nil
+                return []
             }
-            return String(arg.dropFirst(Array.validCommandLineArgSuffix.count))
+            return String(arg.dropFirst(Array.validCommandLineArgSuffix.count)).components(separatedBy: ",")
```
Comment (maxwellE) [inline tools/generator/src/Extensions/Array+Extensions.swift:11]: @rasberik Will this work?
Screenshot 2023-01-06 at 5 37 04 PM

**Review by rasberik — APPROVED** [review]
Works as expected, thank you

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**